addons/web_editor/static/lib/odoo-editor/src/utils/utils.js

```diff
@@ -419,6 +419,13 @@ export function getAdjacents(node, predicate = n => !!n) {
     const next = getAdjacentNextSiblings(node, predicate);
     return predicate(node) ? [...previous.reverse(), node, ...next] : [];
 }
+export function getDescendants(node) {
+    let descendants = [];
+    for (const child of (node.childNodes || [])) {
+        descendants = [...descendants, child, ...getDescendants(child)];
+    }
+    return descendants;
```

Comment:
In order to no recreate a table at each iteration, you could simply do:
```js
return [...(node.childNodes || []), ...getDescendants(child)];
```

Reply:
Euuuuh totally right, wtf did I do here ^^